Bug in plugin on Windows Platform

Wrong line separators: '\r\nModel mo...' at offset 0
java.lang.AssertionError: Wrong line separators: '\r\nModel mo...' at offset 0
	at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:66)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.util.text.StringUtil.assertValidSeparators(StringUtil.java:2518)
	at com.intellij.openapi.editor.impl.DocumentImpl.assertValidSeparators(DocumentImpl.java:635)
	at com.intellij.openapi.editor.impl.DocumentImpl.<init>(DocumentImpl.java:125)
	at com.intellij.openapi.editor.impl.DocumentImpl.<init>(DocumentImpl.java:120)
	at com.intellij.openapi.editor.impl.DocumentImpl.<init>(DocumentImpl.java:116)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createDocument(EditorFactoryImpl.java:130)
	at com.imaginea.kodebeagle.action.RefreshAction.createCodePaneTinyEditor(RefreshAction.java:304)
	at com.imaginea.kodebeagle.action.RefreshAction.buildCodePane(RefreshAction.java:295)
	at com.imaginea.kodebeagle.action.RefreshAction.doFrontEndWork(RefreshAction.java:220)
	at com.imaginea.kodebeagle.action.RefreshAction.access$500(RefreshAction.java:77)
	at com.imaginea.kodebeagle.action.RefreshAction$QueryKBServerTask.onSuccess(RefreshAction.java:582)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1$2.run(ProgressManagerImpl.java:153)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.Throwable
	... 31 more

--- a/plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/SpotlightPane.java
+++ b/plugins/idea/kodebeagleidea/src/main/java/com/imaginea/kodebeagle/ui/SpotlightPane.java
@@ -27,10 +27,10 @@ import com.intellij.ide.BrowserUtil;
 import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.editor.Editor;
 import com.intellij.openapi.editor.EditorFactory;
+import com.intellij.openapi.editor.impl.DocumentImpl;
 import com.intellij.openapi.fileEditor.FileEditorManager;
 import com.intellij.openapi.fileTypes.FileType;
 import com.intellij.openapi.fileTypes.FileTypeManager;
-import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.ui.JBColor;
 
@@ -85,23 +85,21 @@ public class SpotlightPane {
     private void createSpotlightPaneTinyEditor(final JPanel spotlightPaneTinyEditorJPanel,
                                           final String displayFileName, final String fileName,
                                           final String contents) {
-        Document tinyEditorDoc =
-                EditorFactory.getInstance().createDocument(contents);
+        Document tinyEditorDoc = new DocumentImpl(contents, true, false);
         tinyEditorDoc.setReadOnly(true);
-        Project project = windowObjects.getProject();
         FileType fileType =
                 FileTypeManager.getInstance().getFileTypeByExtension(MainWindow.JAVA);
 
         Editor tinyEditor =
                 EditorFactory.getInstance().
-                        createEditor(tinyEditorDoc, project, fileType, false);
+                        createEditor(tinyEditorDoc, windowObjects.getProject(), fileType, false);
         tinyEditor.getContentComponent().addMouseWheelListener(new MouseWheelListener() {
             @Override
             public void mouseWheelMoved(final MouseWheelEvent e) {
                 spotlightPaneTinyEditorJPanel.dispatchEvent(e);
             }
         });
-        windowEditorOps.releaseEditor(project, tinyEditor);
+        windowEditorOps.releaseEditor(windowObjects.getProject(), tinyEditor);
 
         JPanel expandPanel = new JPanel(wrapLayout);
 

